### ISSUE TITLE
NET-1821 Fix S2857 FN: No issue is raised if SQL related namespaces are imported through global using directives

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#5956]
### Description

No issue is raised if SQL related namespaces are imported through global using directives

### Repro steps

```csharp
// global using System.Data.SqlClient is declared in another file.
class Foo
{
    private string field = "SELECT *" + // FN
        "FROM TABLE" +
        "WHERE X = 1;";
}
```

### Expected behavior

Issue is raised.

### Actual behavior

No issue is raised.

### Related information

* C#/VB.NET Plugins version: 8.42.0.51121

### How?

Global using retrieval will be available from MSBuild version 17.3 onwards through [this ](https://github.com/CyrusNajmabadi/roslyn/blob/main/src/Compilers/Core/Portable/Compilation/IImportScope.cs) API,  exposed via `semanticModel.GetImportScopes`.

Since the semantic model will be called, we have to consider the potential performance regression but running tests.

**Comment by mary-georgiou-sonarsource** [comment]
It might be possible to retrieve from Roslyn what usings are in scope at the syntax node that is being analyzed.

Otherwise there might be a big performance impact.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1821](https://sonarsource.atlassian.net/browse/NET-1821)

[NET-1821]: https://sonarsource.atlassian.net/browse/NET-1821?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
